src/octree.jl

```diff
@@ -0,0 +1,154 @@
+
+struct OctreeQuantization <: AbstractColorQuantizer
+    numcolors::Int
+    function OctreeQuantization(
+        colorspace::Type{<:Colorant},
+        numcolors::Int = 256;
+        kwargs...,
+    )
+        colorspace == RGB{N0f8} || error("Octree Algorithm only supports RGB colorspace")
+        return new(numcolors)
+    end
+end
+
+const OCTREE_DEFAULT_COLORSPACE = RGB{N0f8}
+# Color to RGB
+# constructor for struct
+function OctreeQuantization(numcolors::Int = 256; kwargs...)
+    return OctreeQuantization(OCTREE_DEFAULT_COLORSPACE, numcolors; kwargs...)
+end
+
+function (alg::OctreeQuantization)(img::AbstractArray)
+    return octreequantisation!(img; numcolors = alg.numcolors)
```

Comment:
I think I would want to do this and reluctant to provide API for third given incase we do return a function that acts as quantizer, that will compare each pixel with all quantized colors for similarity which might make it unusable but for a single color it would work:
```
img, palette = quantize(img, alg; return_palette=true)
img = quantize(img, alg; return_palette=false)
quantize!(img, alg) # we can return palette here too but looks unusual
```